### DISCUSSION TITLE
log-file / solver non-linear petsc

### DISCUSSION THREAD

**Discussion by cdaversin** [feelpp/feelpp#328] [General]
The informations given in the log files during a non-linear solve can be surprising:

```
solvernonlinearpetsc.cpp:775] iteration 0: Linear iterations : 0 Function norm = 448193
solvernonlinearpetsc.cpp:775] iteration 1: Linear iterations : 1 Function norm = 9.43635
solvernonlinearpetsc.cpp:775] iteration 2: Linear iterations : 1 Function norm = 0.511802
solvernonlinearpetsc.cpp:775] iteration 3: Linear iterations : 1 Function norm = 0.000549241
solvernonlinearpetsc.cpp:775] iteration 4: Linear iterations : -2075081832 Function norm = 3.21143e-322
solvernonlinearpetsc.cpp:775] iteration 5: Linear iterations : 32767 Function norm = 1.28278e-311 
```

Indeed, this is due to the loop in solvernonlinearpetsc.hpp:

```
    for ( int i=0; i<50/*n_iterations+1*/; i++ )
    {
        LOG(INFO) << "iteration " << i << ": Linear iterations : " << hist_its[i] << " Function norm = " << history[i] << "\n";
    }
```

Is there a reason for which this loop is run 50 times and doesn't depend on the real number of iterations needed to converge ?

**Comment by vincentchabannes** [comment]
Le 4 avr. 2014 à 14:35, Cécile Daversin notifications@github.com a écrit :

> The informations given in the log files during a non-linear solve can be surprising:
> 
> solvernonlinearpetsc.cpp:775] iteration 0: Linear iterations : 0 Function norm = 448193
> solvernonlinearpetsc.cpp:775] iteration 1: Linear iterations : 1 Function norm = 9.43635
> solvernonlinearpetsc.cpp:775] iteration 2: Linear iterations : 1 Function norm = 0.511802
> solvernonlinearpetsc.cpp:775] iteration 3: Linear iterations : 1 Function norm = 0.000549241
> solvernonlinearpetsc.cpp:775] iteration 4: Linear iterations : -2075081832 Function norm = 3.21143e-322
> solvernonlinearpetsc.cpp:775] iteration 5: Linear iterations : 32767 Function norm = 1.28278e-311 
> Indeed, this is due to the loop in solvernonlinearpetsc.hpp:
> 
> ```
> for ( int i=0; i<50/*n_iterations+1*/; i++ )
> {
>     LOG(INFO) << "iteration " << i << ": Linear iterations : " << hist_its[i] << " Function norm = " << history[i] << "\n";
> }
> ```
> 
> Is there a reason for which this loop is run 50 times and doesn't depend on the real number of iterations needed to converge ?

The size of history tab is requiert before the solve.
I think that it's better to take the max iteration number for this nlSolve.
